.github/workflows/on-push.yaml

```diff
@@ -0,0 +1,254 @@
+name: On Push
+on: [push]
+
+concurrency:
+  # limit concurrency of entire workflow runs for a specific branch
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  NODE_VERSION: 22.1.0
+  BUF_VERSION: 1.42.0
+  PYTHON_VERSION: 3.13
+  CARGO_INCREMENTAL: 0
+  CARGO_PROFILE_DEV_STRIP: "debuginfo"
+
+jobs:
+
+  # Runs various lints and checks for the project, including Rustfmt, Clippy,
+  # Protobuf lints, and pnpm lints.
+  lint:
+    name: Run Lints
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: rui314/setup-mold@v1
+        with:
+          make-default: true
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+      - uses: pnpm/action-setup@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: "pnpm"
+      - uses: arduino/setup-protoc@v3
+        with:
+          version: "25.3"
+      - uses: bufbuild/buf-action@v1
+        with:
+          version: ${{ env.BUF_VERSION }}
+          setup_only: true
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          components: clippy, rustfmt
+          cache-key: "rust-tests"
+      - name: Install Package Dependencies
+        run: make install
+      - name: Lint (Rustfmt)
+        run: cargo fmt --all -- --check
+      - name: Lint (Clippy)
+        run: cargo clippy -- -D warnings
+      - name: Lint (pnpm)
+        run: pnpm --recursive run lint
+      - name: Lint (Protobuf)
+        run: |
+          buf format --diff --exit-code
+          buf lint
+        working-directory: ./protobufs
+      - name: Typecheck (pnpm)
+        run: pnpm --recursive typecheck
+
+  # Builds the Rust test artifacts for the project, packages them as Nextest
+  # archives and uploads them as artifacts. This job is used as a dependency for
+  # the `unit-tests` and `integration-tests` jobs.
+  build-tests:
+    name: Build Test Artifacts
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: rui314/setup-mold@v1
+        with:
+          make-default: true
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          cache-key: "rust-tests"
+      # - uses: pnpm/action-setup@v4
+      # - uses: actions/setup-node@v3
+      #   with:
+      #     node-version: ${{ env.NODE_VERSION }}
+      #     cache: "pnpm"
+      # - uses: arduino/setup-protoc@v3
+      #   with:
+      #     version: "25.3"
```

Comment:
Why not just delete, same thing for a the commented out `Install Package Dependencies`

Reply:
Yeah should be removed, miss

Reply:
[7d93766](https://github.com/stacks-network/sbtc/pull/1238/commits/7d93766c0ebfc80a7d841f846629944a75d0e0ec)